pydantic/_internal/_typing_extra.py

```diff
@@ -120,7 +105,8 @@ def is_annotated(tp: Any, /) -> bool:
     #> True
     ```
     """
-    return _is_typing_name(get_origin(tp), name='Annotated')
+    origin = get_origin(tp)
+    return origin is t_e.Annotated or (hasattr(typing, 'Annotated') and origin is typing.Annotated)
```

Comment:
Note: this can also be done differently, e.g.:

```python
IS_PY39 = sys.version_info >= (3, 9)

def is_annotated(tp: Any, /) -> bool:
    origin = get_origin(tp)
    return origin is t_e.Annotated or IS_PY39 and origin is typing.Annotated
```

But all the approaches are equivalent in term of performance